File: Java/Aplikasi-Search-Github/app/src/main/java/com/makhalibagas/searchgithub/ViewModel/DetailViewModel.java
package com.makhalibagas.searchgithub.ViewModel;


import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.makhalibagas.searchgithub.Model.User;
import com.makhalibagas.searchgithub.Retrofit.ApiService;
import com.makhalibagas.searchgithub.Retrofit.RetroConfiguration;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class DetailViewModel extends ViewModel {
    private MutableLiveData<User> useritem;

    public void setDataDetail(String username){

        ApiService apiService = RetroConfiguration.getRetrofit().create(ApiService.class);
        Call<User> call = apiService.getUserDetail(username);
        call.enqueue(new Callback<User>() {
            @Override
            public void onResponse(Call<User> call, Response<User> response) {
                useritem.setValue(response.body());
            }

            @Override
            public void onFailure(Call<User> call, Throwable t) {

            }
        });
    }

    public LiveData<User> getDataDetail(){
       if (useritem == null){
           useritem = new MutableLiveData<>();
       }
       return useritem;
    }
}

File: Kotlin/Submission2/app/src/main/java/com/makhalibagas/submission2/viewmodel/DetailViewModel.kt
package com.makhalibagas.submission2.viewmodel

import android.content.Context
import android.widget.Toast
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.makhalibagas.submission2.model.detailUser.DetailUser
import com.makhalibagas.submission2.retrofit.RetroConfig
import retrofit2.Call
import retrofit2.Response

/**
 * Created by Bagas Makhali on 7/3/2020.
 */
class DetailViewModel : ViewModel() {


    private val userDetail : MutableLiveData<DetailUser> = MutableLiveData()

    fun loadUser(context: Context, username:String){
        RetroConfig.getRetrofit().getDetailUser(username).enqueue(object :retrofit2.Callback<DetailUser>{
            override fun onFailure(call: Call<DetailUser>, t: Throwable) {
                Toast.makeText(context, "Please check your internet connection", Toast.LENGTH_SHORT).show()
            }

            override fun onResponse(call: Call<DetailUser>, response: Response<DetailUser>) {
                if (response.body() != null){
                    userDetail.value = response.body()
                }
            }
        })
    }

    val getDetailUser : LiveData<DetailUser> = userDetail
}